device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/edge/MethodRequest.java

```diff
@@ -55,7 +55,7 @@ public MethodRequest(String methodName, String payload) throws IllegalArgumentEx
      * @param connectionTimeout the timeout in seconds for the connection to be established
      * @throws IllegalArgumentException if the provided methodName is null or empty
      */
-    public MethodRequest(String methodName, String payload, Integer responseTimeout, Integer connectionTimeout) throws IllegalArgumentException
+    public MethodRequest(String methodName, Object payload, Integer responseTimeout, Integer connectionTimeout) throws IllegalArgumentException
```

Comment:
Here and everywhere that we take an Object payload, you'll need to add some explanation in the javadocs as to what kinds of objects we allow here. For instance, describe what would happen if the user passed in a complex nested object vs a String vs null